README.md

```diff
@@ -5,100 +5,110 @@ Module to generate migrations based on source CSV files
+
+There are some rules for structure and content of source csv file:
+* Each source file should have first column filled with Ids (it will be used as unique identifier for source row)
+* All other columns should be named with exact field's machine name.
+* For boolean fields you can use "1/0" or "true/false" values.
+* For list fields, we should have key values in source csv, not labels.
 * Fields with multiple values should have strict delimiter. `|` for example.
-* For list fields, we should have keys in source csv, not labels
-* Date fields should have strict date format everywhere.
-* Sometimes, field consists of several values, need to support that too. We can use `/` separator in column name for these cases.
-Example of field types to support:
-  - **formatted text** (we have format and value)
-  - **link**
-  - **datetime_range**
-  - **address**
-  - **price**. 
- 
-* For file migration, we could have 2 possible cases:
-  - we have absolute filepath in csv -> then we use file_copy process plugin
-  - we have URLs in csv -> we use download plugin
 
-   and we could use our plugin https://www.drupal.org/project/migrate_plus/issues/3113394 here too
+  This delimiter can be specified by **values_delimiter** option in drush command.
+* Date fields should have same date format everywhere in sources.
 
-**TODO:** What other specific field types should we support?
+  This date format can be specified by **date_format** option in drush command.
 
-------------
-## Reference fields support
+## Complex fields with multiple properties.
 
-**References is most complicated part here, as we should have migration dependencies for them**
+For complex fields, like Formatter text, Link, Datetime Range, Address, Price, we could have several properties in source file.
+In this case, you can use `/` separator in column name for these cases.
 
-We could handle it next way:
-1. We know, what sources do we have.
-2. We know field machine name from csv header, so from field definition we could know if it is reference field and what entity and bundle is the target
+For example, `body/value`, `body/format`, `price/number`, `price/currency_code`, etc.
 
-So, if we also have a source for this entity and bundle => then **we have migration dependency here**
+Example of supported complex field types:
+  - **formatted text** (we have format and value properties)
+  - **link** (we could have uri and title properties)
+  - **datetime_range** (we could have value and end_value properties)
+  - **address**
+  - **price**. (we could have number and currency_code properties)
 
-**For all reference fields, we should have Id from corresponding source in field's column.**
+Any other complex field type also could work this way, but it is not guaranteed.
 
-For example, if you have related term for content, in `category` column in content source csv, you'd have id of this term and term source should have row with this Id in its source file:
-in node csv :
+## File and Image fields.
```

Comment:
No need for dot here > `## File and Image fields`

Reply:
fixed